### PR TITLE
CLI sync S3 end to end integration test

### PR DESCRIPTION
**Pull request by alishakawaguchi**
completes NEOS-1653

### CODE REVIEW

File: internal/testutil/testdata/jobmapping_generator.go [hunk 1]
```diff
@@ -0,0 +1,362 @@
+package main
```
Comment (alishakawaguchi) [inline internal/testutil/testdata/jobmapping_generator.go:1]: copied from worker integration tests. Will eventually remove the worker one in favor of this one

File: internal/testutil/testdata/postgres/alltypes/create-tables.sql [hunk 2]
```diff
@@ -135,7 +134,7 @@ INSERT INTO alltypes.all_postgres_types (
     DEFAULT,
     32767,  -- smallint_col
     2147483647,  -- integer_col
-    9223372036854775807,  -- bigint_col
+    922337203685477580,  -- bigint_col
```
Comment (alishakawaguchi) [inline internal/testutil/testdata/postgres/alltypes/create-tables.sql:137]: Max postgres bigint is 9223372036854775807 but when written to S3 and then synced to postgres causes the number to be too large. Need to investigate further

File: worker/pkg/integration-test/datasync_workflow.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/metric"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/testsuite"
+)
+
+type Option func(*TestWorkflowEnv)
```
Comment (nickzelei) [inline worker/pkg/integration-test/datasync_workflow.go:38]: fancy

**Review by nickzelei — APPROVED** [review]
:shipit:

File: internal/testutil/testdata/postgres/alltypes/create-tables.sql (hunk 2 above)
Comment (nickzelei) [reply internal/testutil/testdata/postgres/alltypes/create-tables.sql:137]: Hmm, I don't like this.

Comment (alishakawaguchi) [reply internal/testutil/testdata/postgres/alltypes/create-tables.sql:137]: yeah same

**Pull request merged**